test/python/transpiler/test_cx_cancellation.py

```diff
@@ -45,3 +45,73 @@ def test_pass_cx_cancellation(self):
         resources_after = out_circuit.count_ops()
 
         self.assertNotIn("cx", resources_after)
+
+    def test_pass_cx_cancellation_intermixed_ops(self):
+        """Cancellation shouldn't be effected by the order of ops on different qubits."""
+        qr = QuantumRegister(4)
+        circuit = QuantumCircuit(qr)
+        circuit.h(qr[0])
+        circuit.h(qr[1])
+        circuit.cx(qr[0], qr[1])
+        circuit.cx(qr[2], qr[3])
+        circuit.cx(qr[0], qr[1])
+        circuit.cx(qr[2], qr[3])
+
+        pass_manager = PassManager()
+        pass_manager.append(CXCancellation())
+        out_circuit = pass_manager.run(circuit)
+        resources_after = out_circuit.count_ops()
+
+        self.assertNotIn("cx", resources_after)
+
+    def test_pass_cx_cancellation_chained_cx(self):
+        """Include a test were not all operations can be cancelled."""
+        qr = QuantumRegister(4)
+        circuit = QuantumCircuit(qr)
+        circuit.h(qr[0])
+        circuit.h(qr[1])
+        circuit.cx(qr[0], qr[1])
+        circuit.cx(qr[1], qr[2])
+        circuit.cx(qr[0], qr[1])
+        circuit.cx(qr[2], qr[3])
+        circuit.cx(qr[2], qr[3])
+
+        pass_manager = PassManager()
+        pass_manager.append(CXCancellation())
+        out_circuit = pass_manager.run(circuit)
+        resources_after = out_circuit.count_ops()
+
+        self.assertIn("cx", resources_after)
+        self.assertEqual(resources_after["cx"], 3)
```

Comment:
Would it be possible to test against the exact forms of the circuits for simple cases, not just the instruction counts?

Reply:
Sure, probably, but where do you see that having an effect here?

Reply:
For the ones where no change is expected, I can test that instead

Reply:
Ah, I didn't check which test this was. I agree with you

Reply:
Updated all of them! Much better now